blocks/api/post.pegjs

```diff
@@ -17,10 +17,25 @@ WP_Block_List
   = WP_Block*
 
 WP_Block
-  = WP_Block_Void
+  = WP_Tag_More
+  / WP_Block_Void
   / WP_Block_Balanced
   / WP_Block_Html
 
+WP_Tag_More
+  = "<!--" WS* "more" customText:(WS+ text:$((!(WS* "-->") .)+) { return text })? WS* "-->" noTeaser:(WS* WP_Tag_NoTeaser)?
```

Comment:
As mentioned in #1329, `<!--noteaser-->` doesn't need to directly follow the `more` tag (although it makes sense to put it there). It can theoretically be anywhere in the post content. See `get_the_content()`.